app/controllers/v3/application_controller.rb

```diff
@@ -34,6 +34,31 @@ def bad_request!(message)
     raise CloudController::Errors::ApiError.new_from_details('BadRequest', message)
   end
 
+  def is_a_valid_utf_8_string?(item)
+    return false unless item.is_a?(String)
+    return false if item.force_encoding('UTF-8').valid_encoding?
+
+    message_parse_error!('Invalid UTF-8 encoding in YAML data')
+  end
+
+  def utf8_error!(yaml_data)
```

Comment:
As recursive functions like this can run into performance issues, it would be good to test how long this takes if an app manifest with the maximum allowed size has been provided. If it takes too long it could be a potential DoS attack surface.

Reply:
The yaml parsing (Psych.safe_load) takes ~ 0,5 seconds, the utf-8 check takes ~0,02 s for a manifest file with 1MB (~ 39200 lines)